src/errors.ts

```diff
@@ -92,15 +85,37 @@ export class QueryRuntimeError extends ServiceError {
  * failing.
  */
 export class QueryCheckError extends ServiceError {
-  constructor(failure: QueryFailure, httpStatus: 400) {
-    super(failure, httpStatus);
+  constructor(failure: QueryFailure) {
+    super(failure, 400);
     if (Error.captureStackTrace) {
       Error.captureStackTrace(this, QueryCheckError);
     }
     this.name = "QueryCheckError";
   }
 }
 
+/**
+ * An error due to a "compile-time" check of the query
+ * failing.
+ */
+export class QueryAbortError extends ServiceError {
```

Comment:
Maybe just AbortError so it's consistent with openapi? I don't think we should diverge from their naming conventions where possible.

```suggestion
export class AbortError extends ServiceError {
```